rdr_service/dao/genomics_dao.py

```diff
@@ -3537,8 +3645,26 @@ def get_aw3_array_records(self, **kwargs):
         # should be only array genome but query also
         # used for array investigation workflow
         genome_type = kwargs.get('genome_type', config.GENOME_TYPE_ARRAY)
+        return_missing_files = kwargs.get('return_missing_files', False)
         with self.session() as session:
-            return session.query(
+            subquery = session.query(
+                GenomicSetMember.id,
+                func.count(GenomicGcDataFile.file_type).label("file_count")
+            ).join(
+                GenomicGCValidationMetrics,
+                GenomicGCValidationMetrics.genomicSetMemberId == GenomicSetMember.id
+            ).outerjoin(
+                GenomicGcDataFile,
+                GenomicGcDataFile.identifier_value == GenomicGCValidationMetrics.chipwellbarcode
+            ).filter(
+                GenomicSetMember.genomeType == genome_type,
+                GenomicGcDataFile.ignore_flag != 1,
+                GenomicGcDataFile.file_type.in_(
+                    [file_type['file_type'] for file_type in array_file_types_attributes
+                     if file_type['required']])
```

Comment:
maybe we make it an instance variable since this comprehension is used multiple times.